### DISCUSSION TITLE
Would you be open to providing a way to convert email struct into raw format?

### DISCUSSION THREAD

**Discussion by sarthak-kothari** [mnako/letters#60] [Ideas]
Basically you take a raw formatted io.Reader as input and we get the `email` struct. 
If we wanted to manipulate some fields and then convert the struct to a simple string, would it be possible to support that?

great work btw!

**Comment by rorycl** [comment]
I think you could use the underlying net/mail structures for manipulating header fields. Eg:

```go
package main

import (
    "fmt"
    "log"
    "net/mail"
    "os"

    "github.com/mnako/letters"
)

func main() {
    file := "testfiles/topdir/dir1/email2.eml"
    e, err := os.Open(file)
    if err != nil {
        log.Fatalf("open err %v", err)
    }
    message, err := letters.ParseEmail(e)
    // fmt.Printf("%#v\n", message.Headers) // letters.Headers
    fmt.Printf("%v\n", message.Headers.From)

    newFrom := &mail.Address{
        Name:    "Example Name",
        Address: "address@example.com",
    }
    message.Headers.From = []*mail.Address{newFrom}
    fmt.Printf("%v\n", message.Headers.From)
}
```

Converting the `Email` struct "to a simple string" is a bit challenging, but something that could be taken on for particular needs. The `Email` struct has the following structure:

```
type Email struct {
	Headers Headers

	Text         string
	EnrichedText string // See RFC 1523, RFC 1563, and RFC 1896
	HTML         string

	InlineFiles   []InlineFile
	AttachedFiles []AttachedFile
}
```

So you could just use those to make the string output you want.
